mtp/server.go

```diff
@@ -261,16 +258,11 @@ func (s *LVServer) unregisterMotionClient(w *MJPEGResponseWriter) {
 }
 
 func (s *LVServer) HandleSnapshot(w http.ResponseWriter, r *http.Request) {
-	s.controlLock.Lock()
-	writer := NewSnapshotResponseWriter(w)
-	s.snapshotClients[writer] = true
-	s.controlLock.Unlock()
-
-	<-writer.Context.Done()
+	var jpeg []byte
+	jpeg = s.copyFrame()
 
-	s.controlLock.Lock()
-	delete(s.snapshotClients, writer)
-	s.controlLock.Unlock()
+	writer := NewSnapshotResponseWriter(w)
+	writer.Write(jpeg)
```

Comment:
We no longer need a struct `SnapshotResponseWriter`. Let's have all handling done here.

```suggestion
	w.Header().Set("Content-Type", "image/jpeg")
	w.Write(jpeg)
```